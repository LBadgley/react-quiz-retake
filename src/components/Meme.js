import React, { PureComponent } from 'react';
import { HeaderText, FooterText } from './TextSelector';
// import TextSelectoriu from './TextSelector';

export default class Meme extends PureComponent {
  state = {
    headerText: '',
    image: 'https://www.google.com/url?sa=i&source=images&cd=&ved=2ahUKEwiCuJHzzLTiAhXnh1QKHa3BDekQjRx6BAgBEAU&url=https%3A%2F%2Fpreetkamal.com%2F34-very-funny-friday-meme-that-definitely-make-you-smile%2F&psig=AOvVaw2g6JAuHhQLAZoPQR0AcXCa&ust=1558802213391446',
    footerText: ''
  }

  handleChange = ({ target }) => {
    this.setState({ [target.name]: target.value });
  }

  render() {
    const { headerText, image, footerText } = this.state;

    return (
      <div>
        <HeaderText text={headerText} handleChange={this.handleChange}/>
        <img alt="Friday meme">{image}</img>
        <FooterText text={footerText} handleChange={this.handleChange}/>
      </div>
    );
  }
}
