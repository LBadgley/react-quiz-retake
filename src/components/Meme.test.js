import React from 'react';
import { shallow } from 'enzyme';
import Meme from './Meme';

describe('Display Component text', () => {
  it('renders the display', () => {
    const wrapper = shallow(<Meme />);
    expect(wrapper).toMatchSnapshot();
  });
});
