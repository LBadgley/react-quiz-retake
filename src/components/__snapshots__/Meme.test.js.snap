// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display Component text renders the display 1`] = `
<div>
  <Component
    handleChange={[Function]}
    text=""
  />
  <img
    alt="Friday meme"
  >
    https://www.google.com/url?sa=i&source=images&cd=&ved=2ahUKEwiCuJHzzLTiAhXnh1QKHa3BDekQjRx6BAgBEAU&url=https%3A%2F%2Fpreetkamal.com%2F34-very-funny-friday-meme-that-definitely-make-you-smile%2F&psig=AOvVaw2g6JAuHhQLAZoPQR0AcXCa&ust=1558802213391446
  </img>
  <Component
    handleChange={[Function]}
    text=""
  />
</div>
`;
